sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3575,15 +3575,6 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     })
   }
-
-  test("SPARK-33591: null as a partition value") {
-    val t = "part_table"
-    withTable(t) {
-      sql(s"CREATE TABLE $t (col1 INT, p1 STRING) USING PARQUET PARTITIONED BY (p1)")
-      sql(s"INSERT INTO TABLE $t PARTITION (p1 = null) SELECT 0")
-      checkAnswer(sql(s"SELECT * FROM $t"), Row(0, null))
-    }
```

Comment:
A simpler way is to test DROP PARTITION here.

Reply:
New test covers both v1 In-Memory and Hive external catalogs.

Reply:
OK, then please forward port the new test to branch-3.1, as I used this smaller change while backporting.

Reply:
Here it is https://github.com/apache/spark/pull/31331